File: src/main/java/me/itzisonn_/meazy/parser/ast/ProgramUnit.java
package me.itzisonn_.meazy.parser.ast;

import me.itzisonn_.meazy.instruction.InstructionsSet;
import me.itzisonn_.meazy.runtime.environment.Environment;
import org.jspecify.annotations.NullMarked;

@NullMarked
public interface ProgramUnit {
    void emit(InstructionsSet instructionsSet, Environment environment, ProgramUnit parent);
}

File: src/main/kotlin/me/itzisonn_/meazy/parser/ast/ProgramUnit.kt
package me.itzisonn_.meazy.parser.ast

import me.itzisonn_.meazy.instruction.InstructionsSet
import me.itzisonn_.meazy.runtime.environment.Environment

interface ProgramUnit {
    val children: Set<ProgramUnit>

    context(parents: ParentMap)
    fun emit(instructions: InstructionsSet, environment: Environment)
}
